src/changed-files.js

```diff
@@ -56,12 +57,19 @@ async function changedFiles({
       continue;
     }
 
-    for (let file of _changedFiles) {
-      if (exts.length && exts.every(ext => !file.endsWith(`.${ext}`))) {
-        continue;
-      }
+    if (exts.length === 0 && globs.length === 0) {
+      changedFiles.push(... _changedFiles);
+    } else {
+      for (let file of _changedFiles) {
+        if (exts.length && exts.some(ext => file.endsWith(`.${ext}`))) {
+          changedFiles.push(file);
```

Comment:
I think keeping a single `changedFiles.push(file);` would be easier to reason.


```
    for (let file of _changedFiles) {
      let isMatch = false;

      if (!exts.length && !globs.length) {
        isMatch = true;
      } else if (exts.some(ext => file.endsWith(`.${ext}`))) {
        isMatch = true;
      } else if (globs.some(glob => glob.test(file))) {
        isMatch = true;
      }

      if (isMatch) {
        changedFiles.push(file);
      }
    }
```

Reply:
⭐  Agree. Much easier to read.